### PR TITLE
[misc] feat: add ckpt manager in utils

### PR DESCRIPTION
**Pull request by PeterSH6**
- Support FSDPCheckpointManager
- Support hdfs_io import if installed
- Add CI for FSDPCheckpointManager

TODO:
- Will integrate in the next PR

### CODE REVIEW

File: tests/checkpoint/test_fsdp_ckpt.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    lr_scheduler.step()
+    optimizer.zero_grad()
+
+    # Record logits after loaded checkpoint and update
+    with torch.no_grad():
+        logits_after_load = model(input_ids=input_ids2, attention_mask=attention_mask2).logits
+
+    # Step 4: Verify outputs match
+    if rank == 0:
+        torch.testing.assert_close(logits_before_load, logits_after_load, atol=1e-2, rtol=1e-4)
```
Comment (vermouth1992) [inline tests/checkpoint/test_fsdp_ckpt.py]: This should be compared for every rank? Also, it should be bitwise aligned if we toggle deterministic?

Comment (PeterSH6) [reply tests/checkpoint/test_fsdp_ckpt.py]: bitwise align means using default atol and rtol of `torch.testing.assert_close`?

Comment (PeterSH6) [reply tests/checkpoint/test_fsdp_ckpt.py]: Comparing with logits (not loss) should meet bitwise align requirement?

File: tests/checkpoint/test_fsdp_ckpt.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    optimizer.step()
+    lr_scheduler.step()
+    optimizer.zero_grad()
+
+    # Record logits after loaded checkpoint and update
+    with torch.no_grad():
+        logits_after_load = model(input_ids=input_ids2, attention_mask=attention_mask2).logits
+
+    # Step 4: Verify outputs match
+    torch.testing.assert_close(logits_before_load, logits_after_load)
```
Comment (vermouth1992) [inline tests/checkpoint/test_fsdp_ckpt.py]: Shall we change this to bitwise aligned? atol=0.0, rtol=0.0?

File: verl/utils/fs.py [hunk 3]
```diff
@@ -19,7 +19,10 @@
 import tempfile
 import hashlib
 
-from .hdfs_io import copy, makedirs, exists
+try:
+    from hdfs_io import copy, makedirs, exists
```
Comment (vermouth1992) [inline verl/utils/fs.py]: Add a comment for internal usage only

File: verl/utils/checkpoint/fsdp_checkpoint_manager.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                if self.optimizer is not None:
+                    optimizer_state_dict = self.optimizer.state_dict()
+                else:
+                    optimizer_state_dict = None
+                if self.lr_scheduler is not None:
+                    lr_scheduler_state_dict = self.lr_scheduler.state_dict()
+                else:
+                    lr_scheduler_state_dict = None
+
+                state_dict = {
```
Comment (vermouth1992) [inline verl/utils/checkpoint/fsdp_checkpoint_manager.py]: Actually, a better pratice here is to save model/optimizer/extra_state into separate files. This makes checkpoint merge much easier.

Comment (PeterSH6) [reply verl/utils/checkpoint/fsdp_checkpoint_manager.py]: Nice suggestion!

File: verl/utils/checkpoint/checkpoint_manager.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        abs_path = os.path.abspath(self.previous_save_local_path)
+        print(f'Checkpoint manager remove previous save local path: {abs_path}')
+        if not os.path.exists(abs_path):
+            return
+
+        # remove previous local_path
+        shutil.rmtree(abs_path, ignore_errors=True)
+
+    @staticmethod
+    def local_mkdir(path, is_abs=True):
```
Comment (vermouth1992) [inline verl/utils/checkpoint/checkpoint_manager.py]: The name is_abs is confusing. How about is_abspath?

**Review by eric-haibin-lin — APPROVED** [review]

**Review by vermouth1992 — APPROVED** [review]
